src/test/java/uk/gov/hmcts/reform/sscs/service/ServiceHearingsServiceTest.java

```diff
@@ -278,7 +261,174 @@ void testGetServiceLinkedCaseReturnsLinkedCases() throws Exception {
+                        ExcludeDate.builder().value(DateRange.builder().start("2023-07-01").end("2023-08-02").build()).build()
+                    )
+                )
+                .build()
+        );
+
+        given(ccdCaseService.getCaseDetails(String.valueOf(CASE_ID))).willReturn(caseDetails);
+        ServiceHearingRequest request = ServiceHearingRequest.builder()
+            .caseId(String.valueOf(CASE_ID))
+            .build();
+        serviceHearingsService.getServiceHearingValues(request);
+        verify(ccdCaseService, times(1)).updateCaseData(any(SscsCaseData.class), eq(LISTING_ERROR), anyString(), eq("One of the parties unavailability end date is before start date"));
+    }
+
+    @Test
+    void testGetServiceHearingValues_PartiesUnavailabilityStartDateAndEndDateIsNotProvided_ThenSendToUpdateCaseOnly() throws Exception {
+        given(sessionCategoryMaps.getSessionCategory(BENEFIT_CODE, ISSUE_CODE, true, false))
+            .willReturn(new SessionCategoryMap(BenefitCode.PIP_NEW_CLAIM, Issue.DD, false,
+                                               false, SessionCategory.CATEGORY_03, null));
+        given(referenceDataServiceHolder.getSessionCategoryMaps()).willReturn(sessionCategoryMaps);
+        given(venueService.getEpimsIdForVenue(caseData.getProcessingVenue())).willReturn("9876");
+        given(referenceDataServiceHolder.getVenueService()).willReturn(venueService);
+
+        caseDetails.getData().getAppeal().setHearingOptions(
+            HearingOptions.builder()
+                .wantsToAttend("Yes")
+                .excludeDates(
+                    List.of(
+                        ExcludeDate.builder().value(DateRange.builder().build()).build(),
+                        ExcludeDate.builder().value(DateRange.builder().start("").end("").build()).build(),
+                        ExcludeDate.builder().value(DateRange.builder().start(null).end(null).build()).build()
+                    )
+                )
+                .build()
+        );
+
+        given(ccdCaseService.getCaseDetails(String.valueOf(CASE_ID))).willReturn(caseDetails);
+        ServiceHearingRequest request = ServiceHearingRequest.builder()
+            .caseId(String.valueOf(CASE_ID))
+            .build();
+        ServiceHearingValues serviceHearingValues = serviceHearingsService.getServiceHearingValues(request);
+        serviceHearingValues.getParties().stream()
+            .map(PartyDetails::getUnavailabilityRanges)
+            .forEach(o -> o.forEach(p -> {
+                assertThat(p.getUnavailableFromDate()).isNull();
+                assertThat(p.getUnavailableToDate()).isNull();
+            }));
+        verify(ccdCaseService, times(1)).updateCaseData(any(SscsCaseData.class), eq(UPDATE_CASE_ONLY), anyString(), eq("IDs updated for caseDetails due to ServiceHearingValues request"));
+    }
+
+    @ParameterizedTest
+    @MethodSource("partyMemberNamesParameters")
```

Comment:
Please replace this with @CsvSource. It will achieve the same thing, but the arguments will be right here at the start of the test, rather than somewhere else in the test class. Better for clarity on what is being tested.